### PR TITLE
Add try-except for torch_scatter

### LINKED ISSUE #13007: Importing hides underlying error
## Environment info

(Couldn't run due to bug this issue is about but did my best to fill it in)

- `transformers` version: 4.9.1
- Platform: Linux-3.10.0-1160.31.1.el7.x86_64-x86_64-with-centos-7.9.2009-Core
- Python version: 3.7.11
- PyTorch version (GPU?): 1.7.0+cu102 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: no
- Using distributed or parallel set-up in script?: no

### Who can help

No person for general `transformers` issues is listed

## To reproduce

1. Run `pip install torch-scatter==2.0.6 -f https://pytorch-geometric.com/whl/torch-1.7.0+cpu.html` (install version that has a bug)
2. Use a machine with CUDA support.
3. Run `python -c 'from transformers import AutoModelForCausalLM'` to see the following output:

```python
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: cannot import name 'AutoModelForCausalLM' from 'transformers' (venv/lib/python3.7/site-packages/transformers/__init__.py)
```

4. Run `transformers-cli env` to see the following output:

```python
Traceback (most recent call last):
  File "venv/lib/python3.7/site-packages/torch_scatter/__init__.py", line 14, in <module>
    f'{library}_{suffix}', [osp.dirname(__file__)]).origin)
AttributeError: 'NoneType' object has no attribute 'origin'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "venv/bin/transformers-cli", line 5, in <module>
    from transformers.commands.transformers_cli import main
  File "venv/lib/python3.7/site-packages/transformers/commands/transformers_cli.py", line 23, in <module>
    from .run import RunCommand
  File "venv/lib/python3.7/site-packages/transformers/commands/run.py", line 17, in <module>
    from ..pipelines import SUPPORTED_TASKS, TASK_ALIASES, Pipeline, PipelineDataFormat, pipeline
  File "venv/lib/python3.7/site-packages/transformers/pipelines/__init__.py", line 30, in <module>
    from .automatic_speech_recognition import AutomaticSpeechRecognitionPipeline
  File "/venv/lib/python3.7/site-packages/transformers/pipelines/automatic_speech_recognition.py", line 20, in <module>
    from .base import Pipeline
  File "venv/lib/python3.7/site-packages/transformers/pipelines/base.py", line 43, in <module>
    from ..models.auto.modeling_auto import AutoModel
  File "venv/lib/python3.7/site-packages/transformers/models/auto/modeling_auto.py", line 271, in <module>
    from ..tapas.modeling_tapas import (
  File "venv/lib/python3.7/site-packages/transformers/models/tapas/modeling_tapas.py", line 51, in <module>
    from torch_scatter import scatter
  File "venv/lib/python3.7/site-packages/torch_scatter/__init__.py", line 17, in <module>
    raise AttributeError(e)
AttributeError: 'NoneType' object has no attribute 'origin'
```

(I edited the stack trace to remove the parts of the path outside the virtual environment for improved readability.)

I was fortunate to find this easy-to-create stack trace when writing up this issue. It was actually difficult to find out what was the cause. I had to find the failing line in `transformers` (`from torch_scatter import scatter`) in a much more annoying manner instead.

## Expected behavior

Regular importing of the form `from transformers import ...` should display the full stack trace of the underlying error to provide a usable error message for debugging.

I obviously do not expect this underlying error to be fixed, given that it's not part of `transformers`. However, given that a comment near the failure considers `torch_scatter` a `soft dependency`, it might be a good idea to emit a warning when the package fails to import instead of causing the entire `transformers` library to fail. I'm not using a model that uses `torch_scatter` in the first place, so it shouldn't be required this way.

### PR DESCRIPTION
**Pull request by JetRunner**
Add an error message for the CUDA version mismatch of `torch_scatter`.

### CODE REVIEW

**Comment by aphedges** [comment]
I don't think this will work. It's a `RuntimeError` being raised by `torch_scatter`, not an `OSError`. See the specific code at [line 59 of `__init__.py`](https://github.com/rusty1s/pytorch_scatter/blob/2.0.8/torch_scatter/__init__.py#L59). Also, this replaces the existing informative error message from `torch_scatter` with a less informative one.

File: src/transformers/models/tapas/modeling_tapas.py [hunk 1]
```diff
@@ -46,11 +46,17 @@
...
-    from torch_scatter import scatter
-
-logger = logging.get_logger(__name__)
+    try:
+        from torch_scatter import scatter
+    except OSError:
+        logger.info(
+            "`torch_scatter` can't be loaded. It seems you have a `torch_scatter` installed with the wrong CUDA version."
+            "Please try to reinstall it following the instructions here: https://github.com/rusty1s/pytorch_scatter"
+        )
```
Comment (sgugger) [inline src/transformers/models/tapas/modeling_tapas.py:59]: At this state the model won't be usable, so we should use the `error` level and specifically say that the TAPAS model is not usable.

This code should not be executed anymore if there is no `torch_scatter` installed and the user is not trying to load a TAPAS model.

**Review by sgugger — APPROVED** [review]
Thanks for your PR! Let's make sure you have the right error, and I think an info is too weak now that the models import are all decoupled from AutoModel.

**Comment by JetRunner** [comment]
@aphedges Thanks for the note - I have edited the description that does not indicate the association with your issue anymore. Also, the intention of this PR is to simply circumvent the error since in most cases, people just don't use TAPAS but still get blocked by this error.

**Comment by aphedges** [comment]
@JetRunner, thanks for editing the description!

Sorry about my note about `RuntimeError` vs. `OSError` earlier. I think I got confused by the fact that `torch-scatter` explicitly throws a runtime error for some CUDA version mismatches, but the error you're logging here is for a different CUDA version mismatch that doesn't have a good error message. I think I had to google this one, too, so your error message is definitely an improvement.

**Review by LysandreJik — APPROVED** [review]
LGTM, thanks for taking care of it @JetRunner!

**Pull request merged**